components/eventing-controller/upgrade-job/clients/secret/secret.go

```diff
@@ -0,0 +1,86 @@
+package secret
+
+import (
+	"context"
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// Client struct for k8s Secret client
+type Client struct {
+	client dynamic.Interface
+}
+
+// NewClient creates and returns new client for k8s secrets
+func NewClient(client dynamic.Interface) Client {
+	return Client{client}
+}
+
+// ListByMatchingLabels returns the list of k8s secrets in specified namespace and matching labels.
+// or returns an error if it fails for any reason
+func (c Client) ListByMatchingLabels(namespace string, labelSelector string) (*corev1.SecretList, error) {
```

Comment:
Is `ListByMatchingLabels ` tested?

Reply:
yes, its tested in `check_is_beb_enabled_test.go`.